libs/community/langchain_community/document_loaders/parsers/audio.py

```diff
@@ -12,6 +12,144 @@
+            seconds. If the chunk duration is less or equal than the threshold,
+            it will be skipped.
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        *,
+        deployment_id: str,
+        chunk_duration_threshold: float = 0.1,
+        base_url: Optional[str] = None,
+        api_version: Optional[str] = None,
+        language: Union[str, None] = None,
+        prompt: Union[str, None] = None,
+        response_format: Union[
+            Literal["json", "text", "srt", "verbose_json", "vtt"], None
+        ] = None,
+        temperature: Union[float, None] = None,
+    ):
+        self.api_key = api_key
+        self.base_url = base_url or os.environ.get("AZURE_OPENAI_ENDPOINT")
+        self.api_version = api_version or os.environ.get("OPENAI_API_VERSION")
+
+        self.deployment_id = deployment_id
+        self.chunk_duration_threshold = chunk_duration_threshold
+        self.language = language
+        self.prompt = prompt
+        self.response_format = response_format
+        self.temperature = temperature
+
+    @property
+    def _create_params(self) -> Dict[str, Any]:
+        params = {
+            "language": self.language,
+            "prompt": self.prompt,
+            "response_format": self.response_format,
+            "temperature": self.temperature,
+        }
+        return {k: v for k, v in params.items() if v is not None}
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Lazily parse the blob."""
+
+        import io
+
+        try:
+            import openai
+        except ImportError:
+            raise ImportError(
+                "openai package not found, please install it with "
+                "`pip install openai`"
+            )
+        try:
+            from pydub import AudioSegment
+        except ImportError:
+            raise ImportError(
+                "pydub package not found, please install it with " "`pip install pydub`"
+            )
+
+        if is_openai_v1():
+            # api_key optional, defaults to `os.environ['AZURE_OPENAI_API_KEY']`
+            # api_version optional, defaults to `os.environ['OPENAI_API_VERSION']`
+            # azure_endpoint/base_rule optional,
+            # defaults to `os.environ['AZURE_OPENAI_ENDPOINT']`
+            client = openai.AzureOpenAI(
+                api_key=self.api_key,
+                azure_endpoint=self.base_url,
+                api_version=self.api_version,
+            )
+        else:
+            # Set the API key if provided
+            if self.api_key:
+                openai.api_key = self.api_key
+            if self.base_url:
+                openai.base_url = self.base_url
+
+        # Audio file from disk
+        audio = AudioSegment.from_file(blob.path)
+
+        # Define the duration of each chunk in minutes
+        # Need to meet 25MB size limit for Whisper API
```

Comment:
It could be worth making these kinds of limits configurable - as such size limits do have a tendency to change on the service over time.

Reply:
I'm not really sure how I feel about this part of the code. The limit that was set in the OpenAIWhisperParser allows for 20 minutes of content to be parsed:

```
        # Need to meet 25MB size limit for Whisper API
        chunk_duration = 20
        chunk_duration_ms = chunk_duration * 60 * 1000

        # Split the audio into chunk_duration_ms chunks
        for split_number, i in enumerate(range(0, len(audio), chunk_duration_ms)):
```

This doesn't actually enforce the 25 MB limit and 20 minutes of content is also quite a lot, most likely bigger than 25 MB anyway.

Since the Azure Whisper parser has a maximum of 25MB and will throw an error, it may be better just to remove the file chunking/splitting part of the code. The Azure AI Speech has a [batch transcribe](https://learn.microsoft.com/azure/ai-services/speech-service/batch-transcription-create?pivots=rest-api#use-a-whisper-model) which the documentation recommends to use for larger files. Because of this I'm not sure it really fits the purpose of the Azure Whisper model to be splitting larger files in order to transcribe them?

Let me know what you think anyway!